add http cache warming for sites in modal box (xhr request)

buddhaCode opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://developers.shopware.com/contributing/contribution-guideline/).
-->

### 1. Why is this change necessary?

Shop sites in modal boxes are not taken into account for HTTP cache warm up, currently.

### 2. What does this change do, exactly?

PR changes the static site URL provider for cache warming and adds the shop site URL with 'isXHR' GET parameter.

### 3. Describe each step to reproduce the issue or behaviour.

1. Warm up HTTP cache for shop sites
2. Enabled http cache debug option in config.php
```
    'httpcache' => [
        'debug' => true,
    ],
```
3. Open modal box with shop site in it OR call a shop site's URL with `&isXHR=1` parameter added
4. Check HTTP response headers and find, that request isn't from cache (miss, store)

### 4. Please link to the relevant issues (if any).

Plugin with a backport is available under https://github.com/buddhaCode/BucoStaticXhrHttpCache

### 5. Which documentation changes (if any) need to be made because of this PR?

Guess not.

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

shyim commented:
This breaks the getCount method.

buddhaCode commented:
Ups. Right. Fixed that.

buddhaCode commented:
I'm not sure what the problem with the "check-code-style" is?

soebbing commented:
You should be able to just run `./vendor/bin/php-cs-fixer fix -v engine/Shopware/Components/HttpCache/UrlProvider/StaticProvider.php` to have the codestyle being fixed for you

buddhaCode commented:
Great. Thanks. I've fixed the code style.

hlohaus commented on engine/Shopware/Components/HttpCache/UrlProvider/StaticProvider.php:
```diff
@@ -78,38 +78,61 @@ public function getUrls(Context $context, $limit = null, $offset = null)
-            $context
-        );
+        return array_merge(
+            $this->router->generateList(
```
one generateList call would be better

buddhaCode replied:
Done that. Moved code into separate functions to make it more readable.

buddhaCode commented on engine/Shopware/Components/HttpCache/UrlProvider/StaticProvider.php:
```diff
@@ -79,23 +79,12 @@ public function getUrls(Context $context, $limit = null, $offset = null)
-            )),
+            array_filter(
+                array_merge(
+                    array_map('createRequestParameters', $result),
```
Sorry. Fixed it.

buddhaCode commented:
Any idea what I'm doing wrong? I rebased my branch onto upstream/5.6 and changed the target branch of the PR to 5.6. And still, there are changes from you shopware guys?

soebbing commented:
You need to cherry-pick your commit onto the 5.6 branch. Your clone of 5.5 has some commits that haven't been merged into 5.6 yet

buddhaCode commented:
Ok, but for cherry picking I need to create a new branch and therefore a new PR?!?

buddhaCode commented on engine/Shopware/Components/HttpCache/UrlProvider/StaticProvider.php:
```diff
@@ -132,4 +130,39 @@ private function isShopwareLink($link)
+     *
+     * @return array
+     */
+    private function createRequestParameters($custom)
```
Done. But, are you going to add parameter types and return type and remove all phpdoc with 5.6 release for the entire code base?

soebbing replied:
The current code style removes all typehints that does not provide any additional information.

buddhaCode commented:
@shopwareBot ping. Still incomplete?

shyim commented:
Your tests failing :)

buddhaCode commented:
`     [exec]    1) shopware/engine/Shopware/Components/HttpCache/UrlProvider/StaticProvider.php (return_type_declaration, no_blank_lines_after_phpdoc, single_blank_line_before_namespace)`

Okay, what a pity. I can not get a php7.1 environment up and running in the near future to let the code style fixer run. It's not working with php7.0.
